lib/src/modules/macho/parser.rs

```diff
@@ -327,6 +333,22 @@ impl<'a> MachO<'a> {
                 // everything else
             };
         }
+
+        if let Some(ref dyld_info) = macho.dyld_info {
+            let offset = dyld_info.export_off as usize;
+            let size = dyld_info.export_size as usize;
+            if offset < data.len() {
+                let export_data = &data[offset..offset + size];
```

Comment:
`&data[offset..offset + size]` can panic because `size` is read from the file and it can be arbitrarily large

This should be enough, you don't even need to check for `offset < data.len()`:

```rust
if let Some(export_data) = data.get(offset..offset.saturating_add(size)) {
  ...
}
```

Reply:
ah good catch, thank you! fixed it in some other parts as well instead of the data.len() check!